packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -74,12 +74,11 @@ export function parseBlogFileName(
 ): ParsedBlogFileName {
   const dateFilenameMatch = blogSourceRelative.match(DATE_FILENAME_REGEX);
   if (dateFilenameMatch) {
-    const dateString = dateFilenameMatch.groups!.date!;
-    const text = dateFilenameMatch.groups!.text!;
+    const {folder, text, date: dateString} = dateFilenameMatch.groups!;
     // Always treat dates as UTC by adding the `Z`
     const date = new Date(`${dateString}Z`);
     const slugDate = dateString.replace(/-/g, '/');
-    const slug = `/${slugDate}/${text}`;
+    const slug = `/${slugDate}/${folder}${text}`;
```

Comment:
so folder always ends with a /? I think this a bit confusing but why not, as long as the test covers it...

Reply:
Well, not always, as is the case for `team-a-2021-05-12-hey/announcing-docusaurus-two-beta.md` => `/2021/05/12/team-a-hey/announcing-docusaurus-two-beta`... But in terms of spec I can't think of any better heuristic... Maybe the date string should always be preceded by `/`?

Reply:
🤷‍♂️ that seems good enough for now, we'll see if current regexp produce any weird issue